### PR TITLE
New db function: execMany() - execute many statements with detailed results

### PR DESCRIPTION
**Pull request by cpainterwakefield**
Consider this a feature request - I need functionality as implemented here for my particular use case.  So I wrote some code for it - please let me know if this is something you are willing to incorporate into sql-js.  Thanks!

### CODE REVIEW

**Comment by cpainterwakefield** [comment]
Hi, just wondering if anyone has taken a look at this.  These are features I need for my use case; I'd like to get some sense of whether or not you might include in sql-js in some form, either in current version or next.  If in current version great, if in next version, I might fork and use my fork until the next version comes out.  Thanks!

**Comment by Taytay** [comment]
I haven't taken a detailed look at the code yet, and am not particularly qualified to do so, but I have needed something similar, so I appreciate the contribution. I did notice that the statements don't allow for parameters, which would limit its use in my case and _might_ lead to folks doing some dangerous SQL string manipulation to get this to work in their case. Without knowing the code well, I don't know how hard it would be to make it support parameters, but that was the first thing to stood out to me.

**Comment by cpainterwakefield** [comment]
It would be possible to include parameters, and obviously that would be preferable for use cases where SQL injection is a possibility.  The main thing there would be to work out the interface details (how best to provide the parameters to the function - as an array of arrays matching the # of queries, or maybe just an array of parameters to be used in order?)  Once that was decided, I don't think it would be terribly hard to expand the function to use the parameters.

File: src/api.js [hunk 1]
```diff
@@ -780,6 +787,217 @@ Module["onRuntimeInitialized"] = function onRuntimeInitialized() {
...
+                errorAnswer["success"] = false;
+                errorAnswer["error"] = e.toString();
+                errorAnswer["sql"] = thisSql;
+                answer.push(errorAnswer);
+                if (exitOnError) {
+                    stackRestore(stack);
+                    return answer;
+                }
+                // eslint-disable-next-line no-continue
+                continue;
```
Comment (lovasoa) [inline src/api.js:971]: We can avoid this continue too

**Review by lovasoa — CHANGES_REQUESTED** [review]
I made comments on the implementation details of the PR. But my overall feeling on this is that it tries to address a very specific use case, and that the same could be achieved on the user side if we offered a more generic API. We could have a simple Database.iter_statements API that takes a string and returns an iterable over `Statement` objects. This would be more general, and would allow users to replicate this functionality in an idiomatic manner. Users could also bind parameters to their statements as they see fit. And we could additionally create a `normalized_sql` method on `Statement` that wraps sqlite3_normalized_sql.

**Comment by lovasoa** [comment]
Closed in favor of #429

**Pull request closed without merging**